document_merge_service/api/models.py

```diff
@@ -68,6 +68,4 @@ def auto_delete_file_on_change(sender, instance, **kwargs):
         return
 
     if old_file:
-        new_file = instance.template
-        if not old_file == new_file:
-            DefaultStorage().delete(old_file.name)
+        DefaultStorage().delete(old_file.name)
```

Comment:
This is in `pre_save`, so if you update a template without changing the file, it would just delete it. Probably not what we want

Reply:
Ah indeed, I hadn't considered that case.

So we instead need a better way of detecting if a file has changed (`old_file == new_file` is incorrect, but removing the check altogether isn't either).

Reply:
What is the exact problem we're actually trying to solve here?

Reply:
Added a check now to see if the new file is being uploaded (i.e. if it's a subclass of `UploadedFile`). Still feels like a hack, though.

Reply:
> What is the exact problem we're actually trying to solve here?

This can be reproduced as follows:

 1. Create a new template with a file named `foo.docx`
 2. Modify the template and replace it with another file named `foo.docx` (can be the same file, really.
 3. Observe the storage and see that the new file was saved as `foo_XXXXXXX.docx` (with `XXXXXXX` being random characters), but the old file (`foo.docx`) is also still in the storage. However, this file is no longer referenced by any template, i.e. dangling.

But I'll have to check if this is specific to certain types of storages (checking with S3/Minio here) or just general django-storages behaviour.

Reply:
I now completely replaced the equality check on the files with a check if the new file is being uploaded.